app/lib/common/toolkit/html_editor/mentions/mention_menu.dart

```diff
@@ -0,0 +1,299 @@
+  }
+
+  void hide() {
+    // clearing all the things
+    _updateListener?.cancel();
+    _updateListener = null;
+    _menuEntry?.remove();
+    _menuEntry = null;
+  }
+
+  // control functions
+  static void next() {
+    _menu?.provider.next();
+  }
+
+  static void prev() {
+    _menu?.provider.prev();
+  }
+
+  static void selectCurrent() {
+    if (_menu?._selectCurrent() ?? false) {
+      return;
+    }
+    // selectin failed, just hide
+    _menu?.hide();
+  }
+
+  bool _selectCurrent() {
+    final filtered = switch (mentionType) {
+      MentionType.user => ref.read(filteredUserSuggestionsProvider(roomId)),
+      MentionType.room => ref.read(filteredRoomSuggestionsProvider(roomId)),
+    };
+    final selected = max(
+      0, // 0 or higher
+      switch (mentionType) {
+            MentionType.user => ref.read(selecteUserMentionProvider(roomId)),
+            MentionType.room => ref.read(selectedRoomMentionProvider(roomId)),
+          } ??
+          0, // select the first item as fallback
+    );
+    if (selected >= filtered.length) return false; // too long, ignore
+    final item = filtered.keys.elementAt(selected);
+    _select(mentionType, item, filtered[item]);
+    return true;
+  }
+
+  SelectedMentionNotifier get provider => switch (mentionType) {
+    MentionType.user => ref.read(selecteUserMentionProvider(roomId).notifier),
+    MentionType.room => ref.read(selectedRoomMentionProvider(roomId).notifier),
+  };
+
+  String _makeUri(MentionType type, String id) => switch (type) {
+    MentionType.user => 'matrix:u/${id.substring(1)}',
+    MentionType.room => 'matrix:roomid/$id',
+  };
+
+  void _select(MentionType type, String id, String? displayName) {
+    final selection = editorState.selection;
+    if (selection == null) {
+      MentionMenu.dismiss();
+      return;
+    }
+
+    final transaction = editorState.transaction;
+    final node = editorState.getNodeAtPath(selection.end.path);
+    if (node == null) return;
+
+    final text = node.delta?.toPlainText() ?? '';
+    final cursorPosition = selection.end.offset;
+
+    // Find the trigger symbol position by searching backwards from cursor
+    int atSymbolPosition = -1;
+    final mentionTriggers = [userMentionChar, roomMentionChar];
+    for (int i = cursorPosition - 1; i >= 0; i--) {
+      if (mentionTriggers.contains(text[i])) {
+        atSymbolPosition = i;
+        break;
+      }
+    }
+
+    if (atSymbolPosition == -1) return; // No trigger found
+    final lengthToReplace = cursorPosition - atSymbolPosition;
+    final replacementText = displayName ?? id;
+
+    transaction.replaceText(
+      // remove the trigger and content
+      node,
+      atSymbolPosition,
+      lengthToReplace,
+      replacementText,
+      attributes: {'href': _makeUri(type, id), 'inline': true},
+    );
+
+    editorState.apply(transaction);
+    MentionMenu.dismiss();
+  }
+
+  void _show(BuildContext context) {
+    final editorBox = editorState.renderBox;
```

Comment:
Nice, could you also check whether ```GlobalKey``` is being used for the same purpose elsewhere so we can use this access also there and removing global entirely ?